search/search-domain/src/main/java/io/camunda/search/query/ProcessDefinitionQuery.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.search.query;
+
+import io.camunda.search.filter.FilterBuilders;
+import io.camunda.search.filter.ProcessDefinitionFilter;
+import io.camunda.search.page.SearchQueryPage;
+import io.camunda.search.sort.ProcessDefinitionSort;
+import io.camunda.search.sort.SortOptionBuilders;
+import io.camunda.util.ObjectBuilder;
+import java.util.Objects;
+import java.util.function.Function;
+
+public record ProcessDefinitionQuery(
```

Comment:
Maybe you want to exclude `flowNodes` from the document returned by ElasticSearch, but maybe it is that big, isn't?

Reply:
@houssain-barouni If in `ProcessRecordEntity` isn't `flowNodes` defined will it still be fetched?

Reply:
yes, the full document will be fetched, unless you specify something else in `_source ` field filtering.
But it could be overkill to exclude the `flowNodes `, it is up to you :)